mercury-wallet/docker-compose.yml

```diff
@@ -0,0 +1,21 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: mercury-wallet_web_1
+      APP_PORT: 3000
+      
+  web:
+    image: layer2tech/mercury-wallet-ui:latest
+    user: "1000:1000"
+    restart: on-failure
+    stop_grace_period: 1m
+    ports:
+      - 3000:3000
+  walletserver:
+    image: layer2tech/mercury-tor-adapter:latest
+    user: "1000:1000"
+    restart: on-failure
+    ports:
+      - 3001:3001

```

Comment:
Is there a need for this port to be open on the host?

Reply:
This container needs to be accessible on 3001 (can change) by the main app container (web).

Reply:
Hmm, but does the `web` container talk directly to this container? Or would a user access the frontend (in a browser) using port 3000, and that web app talks to this container on port 3001?

Reply:
Sorry for the delay - yes, the latter -  the a user accesses the frontend using port 3000, and the web container talks to this container on port 3001.

Reply:
Ports like 3001 on the host may already be taken by other apps (in this case, Ride The Lightning) https://github.com/getumbrel/umbrel-apps/blob/bc9101e100609ac455f872c608a1883cf031c61d/ride-the-lightning/umbrel-app.yml#L37

In order for this to work nicely I think we should rely on Docker's internal DNS to resolve container names and allow separate containers to talk to each other. 

The `port` value in your `umbrel-app.yml` defines where your app will be accessed on Umbrel. For example, if you set `port: 3005`, then user's will access your app at `http://umbrel.local:3005`. Our App Proxy forwards requests to your `web` container on the port defined in the `app_proxy` container. In your case this is `APP_PORT: 3000`. So even though users access port 3005 on their Umbrel, they get access to the frontend on the web container's 3000 port.

If the frontend `web` container is making HTTP requests and/or websocket connections to the backend `walletserver` container by doing something like `GET http://umbrel.local:3001/some-endpoint`, then instead you could do something like `GET http://walletserver:3001/some-endpoint`. You would not have to map port 3001:3001 in this case. DOcker's internal DNS resolver will resolve `walletserver` to the container's IP address.